libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -129,9 +134,70 @@ void HTTPClient::clear()
 }
 
 
+/**
+ * parsing the url for all needed parameters
+ * @param client Client&
+ * @param url String
+ * @param https bool
+ * @return success bool
+ */
+bool HTTPClient::begin(Client &client, String url) {
+#ifdef KEEP_PRESENT_API
+    _tcpDeprecated.reset(nullptr);
+    _transportTraits.reset(nullptr);
+#endif
+    _tcp = &client;
+
+    // check for : (http: or https:)
+    int index = url.indexOf(':');
+    if(index < 0) {
+        DEBUG_HTTPCLIENT("[HTTP-Client][begin] failed to parse protocol\n");
+        return false;
+    }
+
+    String protocol = url.substring(0, index);
+    if(protocol != "http" && protocol != "https") {
+        DEBUG_HTTPCLIENT("[HTTP-Client][begin] unknown protocol '%s'\n", protocol.c_str());
+        return false;
+    }
+
+    _port = (protocol == "https" ? 443 : 80);
+    return beginInternal(url, protocol.c_str());
```

Comment:
There shouldn't be any need for the protocol anymore, no?  The `WiFiCilent*` passed in should already either be encrypted or not, and the code paths for both are the same, no?

Reply:
Yes it is still needed. The ESP8266HTTPClient is setting up (and maintaining with Keep-Alive) the connection, and it should know what port to use. The idea (also in the original version) of this begin() method , is to pass all the needed information to set up the connection through a fully qualified URL.